controllers/ratelimitpolicy_limits.go

```diff
@@ -108,3 +113,72 @@ func (r *RateLimitPolicyReconciler) buildRateLimitIndex(ctx context.Context, rlp
 
 	return rateLimitIndex, nil
 }
+
+func (r *RateLimitPolicyReconciler) reconcileLimitadorBackRef(limitador *limitadorv1alpha1.Limitador, policyKey client.Object) (updated bool, err error) {
+	policy := client.ObjectKeyFromObject(policyKey)
+	objAnnotations := common.ReadAnnotationsFromObject(limitador)
+	var refs []client.ObjectKey
+
+	val, ok := objAnnotations[common.RateLimitPoliciesBackRefAnnotation]
+	if ok {
+		err := json.Unmarshal([]byte(val), &refs)
+		if err != nil {
+			return false, err
+		}
+		if common.ContainsObjectKey(refs, policy) {
+			r.Logger().V(1).Info("policy references in annotations", "policy", policy)
+			return false, nil
+		}
+	}
+
+	refs = append(refs, policy)
+	serialized, err := json.Marshal(refs)
+	if err != nil {
+		return false, err
+	}
+	objAnnotations[common.RateLimitPoliciesBackRefAnnotation] = string(serialized)
+	limitador.SetAnnotations(objAnnotations)
+	return true, nil
+}
+
+func (r *RateLimitPolicyReconciler) deleteLimitadorBackReference(ctx context.Context, policy client.Object) error {
+	policyKey := client.ObjectKeyFromObject(policy)
+
+	limitadorList, err := r.listLimitadorByNamespace(ctx, policyKey.Namespace)
```

Comment:
The `Limitador` CR is not necessarily in the same namespace as the RLP: https://github.com/Kuadrant/kuadrant-operator/blob/6f3b3fdee2d0213da3331265e14e72a2523cba4f/controllers/ratelimitpolicy_limits.go#L48